packages/shylu/shylu_node/hts/src/shylu_hts_impl_def.hpp

```diff
@@ -4044,41 +4045,47 @@ ondiag_solve (const OnDiagTri& t, Sclr* x, const Int ldx, const Int nrhs,
   const Int nthreads = t.nthreads();
   if (nthreads == 1) {
     t.solve(x, ldx, x, ldx, nrhs);
+#ifdef _OPENMP
+#   pragma omp flush
+#endif
     *t_barrier = step;
```

Comment:
This is the thread-wide lock acquisition?

Reply:
It's not a lock but rather a sync of the thread's local view of memory with the global memory.